projects/solar/test/test_fan_control_solar.c

```diff
@@ -0,0 +1,342 @@
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "max6643_fan_controller.h"
+#include "ms_test_helper_can.h"
+#include "ms_test_helpers.h"
+#include "solar_boards.h"
+#include "solar_events.h"
+#include "spv1020_mppt.h"
+#include "spv1020_mppt_defs.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+// Any defines here
+#define BELOW_TEMPERATURE_THRESHOLD_DC 2
+#define ABOVE_TEMPERATURE_THRESHOLD_DC 4
+#define SPV1020_NOMINAL_MASK 0b0000000
+
+typedef enum {
+  TEST_CAN_SOLAR_FAN_EVENT_TX = 0,
+  TEST_CAN_SOLAR_FAN_EVENT_RX,
+  TEST_CAN_SOLAR_FAN_EVENT_FAULT,
+  NUM_TEST_CAN_EVENTS,
+} TestCanEvent;
+
+// Declare static test variables
+static uint16_t s_full_speed_temp_threshold = 3;  // arbitrarily selected value for testing
+static SolarMpptCount s_mppt_count = SOLAR_BOARD_6_MPPTS;
+
+// Test Board A with 6 MPPTs uses virtual port A
+static GpioAddress s_overtemp_addr = { .port = GPIO_PORT_A, .pin = 0 };
+static GpioAddress s_fan_fail_addr = { .port = GPIO_PORT_A, .pin = 1 };
+static GpioAddress s_full_speed_addr = { .port = GPIO_PORT_A, .pin = 2 };
+
+static GpioState s_gpio_state;
+static Event s_data_ready = { .id = DATA_READY_EVENT, .data = 0 };
+static CanStorage s_can_storage = { 0 };
+static FaultHandlerSettings s_settings = {
+  .relay_open_faults = {},
+  .num_relay_open_faults = 0,
+  .mppt_count = SOLAR_BOARD_6_MPPTS,
+};
+
+// To receive rx_fault_monitor info
+static uint8_t s_ee_solar_fault;
+static uint8_t s_fault_data;
+
+static StatusCode prv_ee_solar_fault_rx_handler(const CanMessage *msg, void *context,
+                                                CanAckStatus *ack_reply) {
+  CAN_UNPACK_SOLAR_FAULT_6_MPPTS(msg, &s_ee_solar_fault, &s_fault_data);
+  return STATUS_CODE_OK;
+}
+
+void setup_test(void) {
+  // Also runs event_queue_init(), gpio_init(), interrupt_init(), soft_timer_init()
+  initialize_can_and_dependencies(&s_can_storage, SYSTEM_CAN_DEVICE_SOLAR_6_MPPTS,
+                                  TEST_CAN_SOLAR_FAN_EVENT_TX, TEST_CAN_SOLAR_FAN_EVENT_RX,
+                                  TEST_CAN_SOLAR_FAN_EVENT_FAULT);
+  gpio_it_init();
+  data_store_init();
+  fault_handler_init(&s_settings);
+
+  // Register Handlers to receive and unpack CAN message into static variables
+  can_register_rx_handler(SYSTEM_CAN_MESSAGE_SOLAR_FAULT_6_MPPTS, prv_ee_solar_fault_rx_handler,
+                          NULL);
+
+  // Used to reinitialize static testing variable GpioState s_gpio_state (default inactive high)
+  s_gpio_state = GPIO_STATE_HIGH;
+}
+
+void teardown_test(void) {}
+
+// Test that "fan fail" faults are raised properly
+void test_fan_fail_fault_handling(void) {
+  FanControlSolarSettings settings = {
+    .overtemp_addr = s_overtemp_addr,
+    .fan_fail_addr = s_fan_fail_addr,
+    .full_speed_addr = s_full_speed_addr,
+    .full_speed_temp_threshold_dC = s_full_speed_temp_threshold,
+    .mppt_count = s_mppt_count,
+  };
+
+  TEST_ASSERT_OK(fan_control_init(&settings));
```

Comment:
Since you do this in almost every test (and for the other two tests it doesn't matter if it's initialized), you can move this to `setup_test`.